backend-shared/exportfuncs.cpp

```diff
@@ -37,33 +38,31 @@ bool ExportCallback::canceled() const
 
 static void exportProfile(ProfileWidget2 *profile, const struct dive *dive, const QString &filename)
 {
+	profile->setProfileState(dive, 0);
 	profile->plotDive(dive, 0, false, true);
-	QImage image = QImage(profile->size() * 4, QImage::Format_RGB32);
+	QImage image = QImage(profile->size(), QImage::Format_RGB32);
 	QPainter paint;
 	paint.begin(&image);
 	profile->render(&paint);
 	image.save(filename);
 }
 
-static ProfileWidget2 *getPrintProfile()
+static std::unique_ptr<ProfileWidget2> getPrintProfile()
 {
-	ProfileWidget2 *profile = MainWindow::instance()->graphics;
-	profile->setToolTipVisibile(false);
+	// Let's say that 800x600 is a "reasonable" profile size. Use four times that for printing.
+	const int scale = 4;
+	QSize size(800 * scale, 600 * scale);
+
+	// TODO: Annoyingly, this still needs a parent window? Otherwise,
+	// the profile is shown as its own window, when calling show() below.
+	auto profile = std::make_unique<ProfileWidget2>(nullptr, MainWindow::instance());
+	profile->resize(size);
+	profile->show();	// Ominous: if the scene isn't shown, parts of the plot are missing. Needs investigation.
```

Comment:
ahhhhhh - ok, so the previous few comments were all just premature. I'll leave them because it's painful to go back when doing the reviews, but now this all makes sense. THANKS